lib/rules/no-undef-components.js

```diff
@@ -130,7 +130,16 @@ module.exports = {
           (scope) => scope.type === 'module'
         )
         for (const variable of (moduleScope && moduleScope.variables) || []) {
-          scriptVariableNames.add(variable.name)
+          const definition = variable.defs[0]
+          if (
```

Comment:
I think we can more easily check if it's a value by doing:

```js
if (!variable.isValueVariable || variable.isValueVariable())
```

https://github.com/typescript-eslint/typescript-eslint/blob/77d76e21cdc2e100c729c839c292e82ab7c554c5/packages/scope-manager/src/variable/Variable.ts#L24

Reply:
This comment of mine was wrong 😓